src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -629,6 +629,49 @@ static std::string parseMqttUrl(ConnectionInfo* ciP, SubscriptionUpdate* subsP,
   return "";
 }
 
+/* ****************************************************************************
+*
+* parseKafkaUrl -
+*/
+static std::string parseKafkaUrl(ConnectionInfo* ciP, SubscriptionUpdate* subsP, const Value& kafka)
+{
+  Opt<std::string> urlOpt = getStringMust(kafka, "url", "url kafka notification");
+
+  if (!urlOpt.ok())
+  {
+    return badInput(ciP, urlOpt.error);
+  }
+  if (!urlOpt.given)
+  {
+    return badInput(ciP, "mandatory kafka field /url/");
+  }
+
+  if (forbiddenChars(urlOpt.value.c_str()))
+  {
+    return badInput(ciP, "forbidden characters in kafka field /url/");
+  }
+
+  std::string  host;
+  int          port;
+  std::string  path;
+  std::string  protocol;
+  if (!parseUrl(urlOpt.value, host, port, path, protocol))
+  {
+    return badInput(ciP, "invalid kafka /url/");
+  }
+  if (protocol != "kafka:")
+  {
+    return badInput(ciP, "http or https URL cannot be used in kafka notifications");
```

Comment:
```suggestion
    return badInput(ciP, "kafka schema is not used in URL");
```

Alternatively, we could use "http or https or mqtt URL cannot be used in kafka notifications" but it scales worse.

Reply:
Fixed in [9b53154](https://github.com/telefonicaid/fiware-orion/pull/4700/commits/9b53154cc4cadb97765364454898352a3b8e3436) PR#4700